python_modules/dagster/dagster/_core/execution/context/system.py

```diff
@@ -199,6 +203,7 @@ class ExecutionData(NamedTuple):
     scoped_resources_builder: ScopedResourcesBuilder
     resolved_run_config: ResolvedRunConfig
     job_def: JobDefinition
+    repository_def_lambda: Callable[[], RepositoryDefinition]
```

Comment:
i left this as a lambda at the moment, maybe overcautiously, to make sure that we don't stuff the whole repo in memory permanently on the context for the duration of the step unless it is specifically asked for within an op or asset